robotoff/spellcheck/base_spellchecker.py

```diff
@@ -0,0 +1,22 @@
+from typing import List
+
+from robotoff.spellcheck.items import SpellcheckItem
+
+
+class BaseSpellchecker:
+    def predict(self, items: List[SpellcheckItem]) -> List[SpellcheckItem]:
+        return [self.predict_one(item) for item in items]
+
+    def predict_one(self, item: SpellcheckItem) -> SpellcheckItem:
+        if item.is_lang_allowed:
+            item.update_correction(
+                correction=self.correct(item.latest_correction), model=self.name
+            )
+        return item
+
+    def correct(self, text: str) -> str:
```

Comment:
A better way to enforce the implementation of the correct method is to declare BaseSpellchecker as an abstract class using the abc module (there are examples in the robotoff project of abstract classes)

Reply:
First time I really use those abstract classes. Changed in https://github.com/openfoodfacts/robotoff/pull/181/commits/4ffc7e554d2b27a5f2fb2170e39859c71d7a4784 . Thanks for the review.